src/vs/base/browser/ui/actionbar/actionbar.ts

```diff
@@ -428,6 +428,8 @@ export class ActionBar extends Disposable implements IActionRunner {
 	protected focusNext(): boolean {
 		if (typeof this.focusedItem === 'undefined') {
 			this.focusedItem = this.viewItems.length - 1;
+		} else if (this.viewItems.length <= 1) {
```

Comment:
I do not like this line, since `ActionBar` has a `preventLoopNavigation` flag. So by default we allow loop navigation. When you are on the last one and press next -> first one is focused

Reply:
Yeah but this is for the case when actionbar has only one action and is already focused, so pressing next won't change the focused action.
This also prevents an infinite loop in the while loop below `while (this.focusedItem !== startIndex ...`